nixos/modules/services/x11/desktop-managers/kde5.nix

```diff
@@ -31,204 +31,224 @@ in
         '';
       };
 
+      extraPackages = mkOption {
+        type = types.listOf types.package;
+        default = [];
+        description = ''
+          KDE packages that need to be installed system-wide.
+        '';
+      };
+
     };
 
   };
 
 
-  config = mkIf (xcfg.enable && cfg.enable) {
+  config = mkMerge [
+    (mkIf (cfg.extraPackages != []) {
+      environment.systemPackages = [ (kde5.kdeWrapper cfg.extraPackages) ];
```

Comment:
I think `kdeWrapper` should be applied to `extraPackages ++ systemPackages` with the latter being the long list of KDE dependencies.